### PR TITLE
Avoid duplicate connections between DlgControllerLearning and MidiController

### PR DESCRIPTION
**Pull request by ferranlala**
The duplicate connections caused `MidiController::commitTemporaryInputMappings` to be called more than once, causing a crash after pressing the "Learn another" button for second time.

### CODE REVIEW

**Comment by ferranpujolcamins** [comment]
2.2.0 is not affected by the crash.

File: src/controllers/dlgcontrollerlearning.cpp [hunk 1]
```diff
@@ -371,6 +371,11 @@ void DlgControllerLearning::commitMapping() {
 }
 
 void DlgControllerLearning::visit(MidiController* pMidiController) {
+    // Disconnect everything so we don't end up with duplicate connections
+    // after pressing the "Learn Another" button
+    pMidiController->disconnect(this);
+    this->disconnect(pMidiController);
```
Comment (uklotzde) [inline src/controllers/dlgcontrollerlearning.cpp]: Explicitly dereferencing this is unusual, but I suppose you did this on purpose in this special case.

Maybe rewording as "Disconnect everything in both directions..." could help to avoid any misinterpretations.

Comment (rryan) [reply src/controllers/dlgcontrollerlearning.cpp]: Is the issue just the double-delivered signal? I'd suggest using Qt::UniqueConnection instead of disconnect.

Comment (uklotzde) [reply src/controllers/dlgcontrollerlearning.cpp]: I also though about this solution. But then you need to take care of every single connection and someone might forget to add this parameter in the future. The safe solution would include both strategies, i.e. disconnect from the controller before connecting and use Qt::UniqueConnection for new connections.

Comment (ferranpujolcamins) [reply src/controllers/dlgcontrollerlearning.cpp]: > Explicitly dereferencing this is unusual, but I suppose you did this on purpose in this special case.

Yes, it felt weird to me otherwise. Lack of symmetry :)

Are you okay to leave it like this? I've changed the comment.

**Comment by ferranpujolcamins** [comment]
Can you merge this? Or is there something else I must address?

**Comment by ferranpujolcamins** [comment]
gentle ping

**Comment by daschuer** [comment]
LGTM, Thank you.
